crates/bevy_render/src/render_phase/mod.rs

```diff
@@ -395,74 +463,62 @@ where
     }
 }
 
-impl UnbatchableBinnedEntityBufferIndex {
+impl UnbatchableBinnedEntityIndexSet {
     /// Adds a new entity to the list of unbatchable binned entities.
-    pub fn add<T>(&mut self, gpu_array_buffer_index: GpuArrayBufferIndex<T>)
-    where
-        T: ShaderSize + WriteInto + Clone,
-    {
-        match (&mut *self, gpu_array_buffer_index.dynamic_offset) {
-            (UnbatchableBinnedEntityBufferIndex::NoEntities, None) => {
-                // This is the first entity we've seen, and we're not on WebGL
-                // 2. Initialize the fast path.
-                *self = UnbatchableBinnedEntityBufferIndex::NoDynamicOffsets {
-                    instance_range: gpu_array_buffer_index.index
-                        ..(gpu_array_buffer_index.index + 1),
+    pub fn add(&mut self, indices: UnbatchableBinnedEntityIndices) {
+        match self {
+            UnbatchableBinnedEntityIndexSet::NoEntities => {
+                if indices.extra_index.is_dynamic_offset() {
+                    // This is the first entity we've seen, and we're on WebGL 2.
+                    // Initialize an array.
+                    *self = UnbatchableBinnedEntityIndexSet::Dense(vec![indices]);
+                } else {
+                    // This is the first entity we've seen, and we're not on WebGL
+                    // 2. Initialize the fast path.
```

Comment:
Could also be on native with GLES or so I suppose? I'd say comment the feature that is present or not, suggesting WebGL2 as an example. So maybe 'we have compute shaders' rather than 'we're not on WebGL2'

Reply:
That comment was there before. This PR just moved things around. I also got surprised by the mention of webgl2 but that's not this PRs fault. Although, still worth updating it I think.

Reply:
I reworded the comment.